crates/net/network/src/network.rs

```diff
@@ -247,18 +248,38 @@ impl NetworkInfo for NetworkHandle {
     fn is_syncing(&self) -> bool {
         SyncStateProvider::is_syncing(self)
     }
+
+    fn is_initially_syncing(&self) -> bool {
+        SyncStateProvider::is_initially_syncing(self)
+    }
 }
 
 impl SyncStateProvider for NetworkHandle {
     fn is_syncing(&self) -> bool {
         self.inner.is_syncing.load(Ordering::Relaxed)
     }
+    // used to guard the txpool
+    fn is_initially_syncing(&self) -> bool {
+        let init_sync_done = self.inner.initial_sync_done.load(Ordering::Relaxed);
+        let is_currently_syncing = self.inner.is_syncing.load(Ordering::Relaxed);
+        is_currently_syncing && !init_sync_done
+    }
 }
 
 impl NetworkSyncUpdater for NetworkHandle {
     fn update_sync_state(&self, state: SyncState) {
-        let is_syncing = state.is_syncing();
-        self.inner.is_syncing.store(is_syncing, Ordering::Relaxed)
+        let future_state = state.is_syncing();
+        let curr_syncing = self.inner.is_syncing.load(Ordering::Relaxed);
+        self.inner.is_syncing.store(future_state, Ordering::Relaxed);
+        // explicitly throw this error away since we expect it to error every time after it's been
+        // set to true, i.e after every live sync
+        let _ = self.inner.initial_sync_done.compare_exchange(
+            false,
+            // on first move back from Syncing->Idle - swap and set init_sync_done to true
+            curr_syncing && !future_state,
+            Ordering::Relaxed,
+            Ordering::Relaxed,
+        );
```

Comment:
this is not quite exactly what I had in mind, you can do a swap on line 272 and then update is_initial_done depending on the values

Reply:
ah.. ok yeah. I misunderstood the initial comment oops, but Ill keep the CAS bc it looks cleaner (and setting `init_sync_done` it in one atomic step is nice)

basically what you're saying is instead of the load/store on `is_syncing` I just need one swap and the return value is what is being set as `curr_syncing` (which I'm going to rename to `prev_state` to make a bit clearer)